### PR TITLE
Fix bugs in the COM refactor of marine DA

### PR DESCRIPTION
**Pull request by guillaumevernieres**
Fixes a couple bugs and does a little cleanup of the COM refactor for marine DA. COM variable definitions are also updated to match the style used in other scripts.

### CODE REVIEW

**Review by WalterKolczynski-NOAA — APPROVED** [review]

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_PREP [hunk 1]
```diff
@@ -22,10 +22,10 @@ export GPREFIX="${GDUMP}.t${gcyc}z."
 export APREFIX="${CDUMP}.t${cyc}z."
 
 # Generate COM variables from templates
-RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx COM_OCEAN_HISTORY:COM_OCEAN_HISTORY_TMPL
-RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx COM_ICE_HISTORY:COM_ICE_HISTORY_TMPL
-RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx COM_ICE_RESTART:COM_ICE_RESTART_TMPL
-
+RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx COM_OCEAN_HISTORY
+RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx COM_ICE_HISTORY
+RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx COM_ICE_RESTART
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_PREP]: I don't think is properly used.  We should use the `_PREV` to use the previous cycle paths.
The proper way to use this is:
```
RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx COM_OCEAN_HISTORY_PREV:COM_OCEAN_HISTORY_TMPL
```

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT [hunk 2]
```diff
@@ -23,8 +23,8 @@ export GPREFIX="${GDUMP}.t${gcyc}z."
 export APREFIX="${CDUMP}.t${cyc}z."
 
 # Generate COM variables from templates
-RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx COM_ATMOS_HISTORY:COM_ATMOS_HISTORY_TMPL
-RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx COM_ATMOS_ANALYSIS:COM_ATMOS_ANALYSIS_TMPL
+RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx COM_ATMOS_HISTORY
+RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx COM_ATMOS_ANALYSIS
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT]: This is not the right way to use previous cycle paths.

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST [hunk 3]
```diff
@@ -14,7 +14,7 @@ export CDATE=${CDATE:-${PDY}${cyc}}
 export GDUMP=${GDUMP:-"gdas"}
 
 # Generate COM variables from templates
-RUN=${GDUMP} YMD=${PDY} HH=${cyc} generate_com -rx COM_OCEAN_ANALYSIS:COM_OCEAN_ANALYSIS_TMPL
+RUN=${GDUMP} YMD=${PDY} HH=${cyc} generate_com -rx COM_OCEAN_ANALYSIS
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST]: Why is this `GDUMP` and mixed with `current_cycle` `PDY` and `cyc`?

**Review by aerorahul — CHANGES_REQUESTED** [review]
The use of COM variables is incorrect.
See inline comments.

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT (hunk 2 above)
Comment (WalterKolczynski-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT]: Sorry, must've missed this when these were first added in the previous PR. Rahul is correct; for consistency, the variable name should be `COM_ATMO_HISTORY_PREV`, etc., which means you will also still need to explicitly provide the template.

**Comment by guillaumevernieres** [comment]
Thanks for the reviews @aerorahul and @WalterKolczynski-NOAA . This work was sloppy, apologies. Looking at what @RussTreadon-NOAA did.

**Review by aerorahul — APPROVED** [review]
looks correct now.

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST [hunk 4]
```diff
@@ -42,6 +41,6 @@ status=$?
 # Remove the Temporary working directory
 ##########################################
 cd "${DATAROOT}" || exit 1
-[[ "${KEEPDATA}" = "NO" ]] && rm -rf "${DATA}"
```
Comment (WalterKolczynski-NOAA) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST]: Can you remove this line if it is no longer active?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST (hunk 4 above)
Comment (guillaumevernieres) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST]: Still used @WalterKolczynski-NOAA , just forgot to revert.

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT [hunk 5]
```diff
@@ -23,8 +23,8 @@ export GPREFIX="${GDUMP}.t${gcyc}z."
 export APREFIX="${CDUMP}.t${cyc}z."
 
 # Generate COM variables from templates
-RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx COM_ATMOS_HISTORY:COM_ATMOS_HISTORY_TMPL
-RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx COM_ATMOS_ANALYSIS:COM_ATMOS_ANALYSIS_TMPL
+RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx COM_ATMOS_HISTORY_PREV:COM_ATMOS_HISTORY_TMPL \
+                                                     COM_ATMOS_ANALYSIS_PREV:COM_ATMOS_ANALYSIS_TMPL
```
Comment (guillaumevernieres) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT]: not my day ... I got that wrong too. I don't need the analysis from the previous cycle.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Comment by WalterKolczynski-NOAA** [comment]
Should I hold this for @AndrewEichmann-NOAA to review or proceed?

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_RUN [hunk 6]
```diff
@@ -6,11 +6,14 @@ export DATA="${DATAROOT}/${RUN}ocnanal_${cyc}"
 source "${HOMEgfs}/ush/jjob_header.sh" -e "ocnanalrun" -c "base ocnanal ocnanalrun"
 
 
+##############################################
+# Set variables used in the script
+##############################################
+
```
Comment (AndrewEichmann-NOAA) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_RUN:12]: Is there something missing here?

Comment (guillaumevernieres) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_RUN:12]: nope

**Review by AndrewEichmann-NOAA — APPROVED** [review]

**Pull request merged**